packages/astro/src/i18n/index.ts

```diff
@@ -161,19 +166,20 @@ export function getPathByLocale(locale: string, locales: Locales) {
  * @param path
  * @param locales
  */
-export function getLocaleByPath(path: string, locales: Locales): string | undefined {
+export function getLocaleByPath(path: string, locales: Locales): string {
 	for (const locale of locales) {
 		if (typeof locale !== 'string') {
 			if (locale.path === path) {
 				// the first code is the one that user usually wants
 				const code = locale.codes.at(0);
+				if (code === undefined) throw new Error("Unreachable");
 				return code;
 			}
 		} else if (locale === path) {
 			return locale;
 		}
 	}
-	return undefined;
+	throw new Error("Unreachable");
```

Comment:
Added errors in a couple of places where implementation did not match the public API.

Reply:
I suggest inserting a better message here. Even though it's for us, it's good practice to leave a message explaining why this happened, or a possible fix.

Reply:
Added in https://github.com/withastro/astro/pull/9790/commits/4abc42ec05bc0ef5fb1770ad3e04b2868fbeb5c6